clang-tools-extra/include-cleaner/lib/Record.cpp

```diff
@@ -232,6 +232,17 @@ class PragmaIncludes::RecordPragma : public PPCallbacks, public CommentHandler {
   void checkForExport(FileID IncludingFile, int HashLine,
                       std::optional<Header> IncludedHeader,
                       OptionalFileEntryRef IncludedFile) {
+    auto AddExport = [&] {
+      auto ExportingFileName = SM.getFileEntryForID(IncludingFile)->getName();
+      if (IncludedFile) {
```

Comment:
if we inline the lambda, we just need these two `if`s, and `ExportingFileName` would be `Top.Path`.

Reply:
i'd rather use the name from file entry, and get rid of `Path` stored in Top (in a follow-up change), as it's not needed anymore.

which later on should bring us 1-step closer to making `PragmaIncludes` copyable, so that clangd can easily make a copy and extend the one from Preamble build, with pragmas inside the main-file AST.